### PR TITLE
feat: upgrade docker environment to Ubuntu 22.04

### LINKED ISSUE #27: update to Ubuntu 22
VSCode will stop supporting Ubuntu 18 in 2025, so we should just upgrade now

### PR DESCRIPTION
**Pull request by alesparza**
We will be doing this course wide

### CODE REVIEW

**Comment by alesparza** [comment]
Please pull this from dockerhub using the tag ubuntu-22
```
docker pull cis548/docker-env:ubuntu-22
docker run -it -v ${pwd}:/home/cit595 --name ubuntu-test cis548/docker-env:ubuntu-22
```
This should automate pulling the image, naming it ubuntu-test, and starting it immediately.  

Please check that clang and gcc work on some program and also run the `test.sh` script that is part of the dovker-env repository (clone repo first).

I did check for W11 ARM but I need someone with the amd architecture (Apple M1 chip) (two separate images)

**Comment by kneruda** [comment]
I can check with an M1 tonight. Sent from my iPhoneOn 31 May 2024, at 4:09 AM, Antonio Esparza ***@***.***> wrote:﻿
Please pull this from docerhub using the tag ubuntu-22
docker pull cis548/docker-env:ubuntu-22
docker run -it -v ${pwd}:/home/cit595 --name ubuntu-test cis548/docker-env:ubuntu-22

This should automate pulling the image, naming it ubuntu-test, and starting it immediately.
Please check that clang and gcc work on some program and also run the test.sh script that is part of the dovker-env repository (clone repo first).
I did check for W11 ARM but I need someone with the amd architecture (Apple M1 chip) (two separate images)

**Comment by nate285** [comment]
Worked for me when I pulled it

**Comment by nate285** [comment]
And the test script worked as well. I'm on x86 ubuntu host machine

File: docker-setup.sh [hunk 1]
```diff
@@ -27,17 +27,21 @@ apt-get install -y tmux
 apt-get install -y gdb valgrind
 
 # install C stuff
-apt-get install -y gcc clang-10 clang-format-10 make
-update-alternatives --install /usr/bin/clang clang /usr/bin/clang-10 100
-update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-10 100
+apt-get install -y gcc clang-16 clang-format-16 make
```
Comment (nate285) [inline docker-setup.sh]: Everything looks good, I was wondering if we had to specify the clang version but when I try and update it, it already says it is on the newest version so I am assuming it does not matter.

**Review by nate285 — COMMENTED** [review]
Once we confirm it is working on arm I think it is all set

**Comment by nate285** [comment]
Actually, is there a way to run the open CI stuff on this and ensure that all the tests work? It should not make a difference but it feels like it is probably a good idea.

File: docker-setup.sh (hunk 1 above)
Comment (alesparza) [reply docker-setup.sh]: it should just be clang and clang-format

**Comment by alesparza** [comment]
> Actually, is there a way to run the open CI stuff on this and ensure that all the tests work? It should not make a difference but it feels like it is probably a good idea.

We could set something up for CI.  Not sure what we would do aside from maybe some basic compilations.  There is a a test.sh that checks that everything installs correctly

**Comment by alesparza** [comment]
The latest image as of right now is the one to verify

**Pull request closed without merging**